paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/lookup/PartitionLoader.java

```diff
@@ -128,12 +128,22 @@ public static PartitionLoader of(FileStoreTable table) {
         }
 
         if (maxPartitionNum == -1) {
+            if (scanPartitions.contains(MAX_PT)) {
+                Duration refresh =
+                        options.get(
+                                FlinkConnectorOptions.LOOKUP_DYNAMIC_PARTITION_REFRESH_INTERVAL);
+                return new DynamicPartitionLevelLoader(
+                        table,
+                        refresh,
+                        ParameterUtils.parseCommaSeparatedKeyValues(scanPartitions));
+            }
```

Comment:
Would it be better to somehow reorganize and improve the readability of this code block? The current organization is seemingly showing the following points:
1. `DynamicPartitionLevelLoader` is closer to `StaticPartitionLoader` than `StaticPartitionLoader`.
2. When `DynamicPartitionLevelLoader` is used, `maxPartitionNum == -1`. i.e. no max_pt() is used.

Both these points seems weird.